deps/rabbit/src/rabbit_classic_queue_index_v2.erl

```diff
@@ -1277,8 +1278,8 @@ queue_name_to_dir_name(#resource { kind = queue,
     rabbit_misc:format("~.36B", [Num]).
 
 segment_file(Segment, #qi{ dir = Dir }) ->
-    filename:join(rabbit_file:binary_to_filename(Dir),
-                  integer_to_list(Segment) ++ ?SEGMENT_EXTENSION).
+    N = integer_to_binary(Segment),
+    <<Dir/binary, N/binary, <<?SEGMENT_EXTENSION>>/binary>>.
```

Comment:
You can just do `<<Dir/binary, N/binary, ?SEGMENT_EXTENSION>>` it will be just as clear.